pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -569,18 +569,17 @@ func (c *Controller) getProxyServiceInstancesByEndpoint(endpoints v1.Endpoints,
 					continue
 				}
 
-				// There is only one IP for kube registry
-				proxyIP := proxy.IPAddresses[0]
-
-				if hasProxyIP(ss.Addresses, proxyIP) {
-					out = append(out, c.getEndpoints(proxyIP, port.Port, svcPort, svc))
-				}
+				for _, ip := range proxy.IPAddresses {
+					if hasProxyIP(ss.Addresses, ip) {
```

Comment:
How could the endpoint contain the ips which are not podip?

Reply:
@hzxuzhonghu  I'm not sure if the logic here is necessary, if it is, I'll add it

Reply:
@hzxuzhonghu It has been modified according to your suggestion